tests/test_cli_compile.py

```diff
@@ -362,38 +362,65 @@ def test_input_file_without_extension(runner):
     assert os.path.exists("requirements.txt")
 
 
-def test_upgrade_packages_option(runner):
+@pytest.mark.parametrize("existing_requirements_txt", [True, False])
+def test_upgrade_packages_option(runner, existing_requirements_txt):
     """
     piptools respects --upgrade-package/-P inline list.
     """
     with open("requirements.in", "w") as req_in:
         req_in.write("small-fake-a\nsmall-fake-b")
-    with open("requirements.txt", "w") as req_in:
-        req_in.write("small-fake-a==0.1\nsmall-fake-b==0.1")
+    if existing_requirements_txt:
+        with open("requirements.txt", "w") as req_in:
+            req_in.write("small-fake-a==0.1\nsmall-fake-b==0.1")
 
     out = runner.invoke(cli, ["-P", "small-fake-b", "-f", MINIMAL_WHEELS_PATH])
 
     assert out.exit_code == 0
-    assert "small-fake-a==0.1" in out.output
+    if existing_requirements_txt:
+        assert "small-fake-a==0.1" in out.output
     assert "small-fake-b==0.3" in out.output
 
 
-def test_upgrade_packages_version_option(runner):
+@pytest.mark.parametrize("existing_requirements_txt", [True, False])
+def test_upgrade_packages_version_option(runner, existing_requirements_txt):
     """
     piptools respects --upgrade-package/-P inline list with specified versions.
     """
     with open("requirements.in", "w") as req_in:
         req_in.write("small-fake-a\nsmall-fake-b")
-    with open("requirements.txt", "w") as req_in:
-        req_in.write("small-fake-a==0.1\nsmall-fake-b==0.1")
+    if existing_requirements_txt:
+        with open("requirements.txt", "w") as req_in:
+            req_in.write("small-fake-a==0.1\nsmall-fake-b==0.1")
 
     out = runner.invoke(cli, ["-P", "small-fake-b==0.2", "-f", MINIMAL_WHEELS_PATH])
 
     assert out.exit_code == 0
-    assert "small-fake-a==0.1" in out.output
+    if existing_requirements_txt:
+        assert "small-fake-a==0.1" in out.output
```

Comment:
Here also need to check `small-fake-a` version if `not existing_requirements_txt`.